config-model/src/test/java/com/yahoo/searchdefinition/processing/DictionaryTestCase.java

```diff
@@ -0,0 +1,138 @@
+// Copyright Verizon Media. Licensed under the terms of the Apache 2.0 license. See LICENSE in the project root.
+
+package com.yahoo.searchdefinition.processing;
+
+import com.yahoo.searchdefinition.Search;
+import com.yahoo.searchdefinition.SearchBuilder;
+import com.yahoo.searchdefinition.derived.AttributeFields;
+import com.yahoo.searchdefinition.document.Dictionary;
+import com.yahoo.searchdefinition.parser.ParseException;
+import com.yahoo.vespa.config.search.AttributesConfig;
+import org.junit.Test;
+
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.fail;
+
+/**
+ * Test configuration of dictionary control.
+ *
+ * @author baldersheim
+ */
+public class DictionaryTestCase {
+    private static AttributesConfig getConfig(Search search) {
+        AttributeFields attributes = new AttributeFields(search);
+        AttributesConfig.Builder builder = new AttributesConfig.Builder();
+        attributes.getConfig(builder);
+        return builder.build();
+    }
+    @Test
+    public void testDefaultDictionarySettings() throws ParseException {
+        String def =
+                "search test {\n" +
+                "    document test {\n" +
+                "        field s1 type string {\n" +
+                "            indexing: attribute | summary\n" +
+                "        }\n" +
+                "\n" +
+                "        field n1 type int {\n" +
+                "            indexing: summary | attribute\n" +
+                "        }\n" +
+                "    }\n" +
+                "}\n";
+        SearchBuilder sb = SearchBuilder.createFromString(def);
+        Search search = sb.getSearch();
+        assertEquals(Dictionary.Type.BTREE, search.getAttribute("s1").getDictionary().getType());
+        assertEquals(Dictionary.Type.BTREE, search.getAttribute("n1").getDictionary().getType());
+    }
+    @Test
+    public void testNumericBtreeSettings() throws ParseException {
+        String def =
+                "search test {\n" +
+                "    document test {\n" +
+                "        field n1 type int {\n" +
+                "            indexing: summary | attribute\n" +
+                "            attribute:fast-search\n" +
+                "            dictionary:btree\n" +
+                "        }\n" +
+                "    }\n" +
+                "}\n";
+        SearchBuilder sb = SearchBuilder.createFromString(def);
```

Comment:
Consider adding a helper function to create the search object and perform the asserts as the same is duplicated in the next two tests.

Reply:
Fixed